create-event-detectors.js

```diff
@@ -0,0 +1,549 @@
+            verbose(`WARNING: Exact case match failed for ${actual} and ${expected}. Tested with toUpperCase() and passed validation.`);
+            return
+        }
+        else
+        throw new Error(`${message}: ${actual} is NOT equal to ${expected}!`)
+    }
+}
+
+function readCsv(fileStore, filePath) {
+    const path = services.fileStoreService.getPathForRead(fileStore, filePath);
+    const lines = Array.from(Files.readAllLines(path));
+    const header = lines.shift().replace(/["']/g, '').replace(/\s+/g, "").split(',');
+
+    //Validate headers 
+    for (let index = 0; index < mainHeaders.length; index++) {
+        compare(header[index], mainHeaders[index], `Fatal Error. Expected column header name ${mainHeaders[index]} mismatch`);
+        //If this is an upper case match, we need to update the mainHeaders array
+        if (header[index].toUpperCase() === mainHeaders[index].toUpperCase()) {
+            header[index] = mainHeaders[index];
+        }
+    }
+
+    //Get the data
+    for (let i = 0; i < lines.length; i++) {
+        const row = lines[i].replace(/["']/g, "").split(',');
+        const data = lines[i] = {};
+        for (let j = 0; j < row.length; j++) {
+            data[header[j]] = row[j];
+        }
+    }
+    return lines;
+}
+
+function assureNotEmpty(name, value) {
+    if (!value) {
+        throw new Error(`Missing required value: ${name}. Validation failed.`);
+    }
+    else if (value == RESERVED_EMPTY) {
+        throw new Error(`Missing required value: ${name}. Value set to ${RESERVED_EMPTY}. Validation failed.`);
+    }
+    return;
+}
+
+const isTypeSupported = (type) => {
+    switch (type) {
+        case "HIGH_LIMIT":
+        case "LOW_LIMIT":
+        case "MULTISTATE_STATE":
+        case "BINARY_STATE":
+        case "NO_UPDATE":
+        case "UPDATE":
+            verbose(`${type}: supported`);
+            return;
+        default:
+            if (type) throw new Error(`detectorType ${type}: Not Supported!`);
+            else throw new Error(`: No supported detectorType provided!`);
+    }
+}
+
+const verbose = (logMessage) => {
+    if (!enableConsoleLog) return;
+    else console.log(logMessage);
+}
+
+const eventDetectorsService = services.eventDetectorsService;
+const eventDetectorsArray = handlersLinkDelimiter === ',' ? [] : readCsv(fileStorePath, fileName);
+const emptyMessage = "[]";
+
+console.log(`Creating ${eventDetectorsArray.length} event detectors`);
+let count = 0;
+let failed = 0;
+
+for (const eventDetectorCsv of eventDetectorsArray) {
+
+    verbose('Begin detector...');
+
+    let detector;
+    let point;
+    let isCreated = true;
+    let insertED = true;
+
+    /*
+    * VALIDATION PATTERN
+    * Perform the validation but do not set the value in the detector.
+    * The value will be set in the detector after it is created.
+    */
+
+    //Validations
+    /*
+    //Why are we checking every value and failing if any of them are missing?

```

Comment:
This comment block should have been removed. There were some validations that were incorrect from the original script that were wrong. I will remove this block.